src/components/lib/cos_kernel_api.c

```diff
@@ -557,6 +612,33 @@ void *
 cos_page_bump_alloc(struct cos_compinfo *ci)
 { return (void*)__page_bump_alloc(ci); }
 
+capid_t
+cos_cap_cpy(struct cos_compinfo *dstci, struct cos_compinfo *srcci, cap_t srcctype, capid_t srccap)
+{
+	capid_t dstcap;
+
+	assert(srcci && dstci);
+
+	dstcap = __capid_bump_alloc(srcci, srcctype);
+	if (!dstcap) return 0;
+
+	if (call_cap_op(srcci->captbl_cap, CAPTBL_OP_CPY, srccap, dstci->captbl_cap, dstcap, 0))  BUG();
+
+	return dstcap;
+}
+
+int
+cos_cap_cpy_at(struct cos_compinfo *dstci, capid_t dstcap, struct cos_compinfo *srcci, capid_t srccap)
+{
+	assert(srcci && dstci);
+
+	if (!dstcap) return 0;
+
+	if (call_cap_op(srcci->captbl_cap, CAPTBL_OP_CPY, srccap, dstci->captbl_cap, dstcap, 0))  BUG();
```

Comment:
I guess this API copies the srccap at any arbitrary location in dstci passed in as dstcap. So, if the caller passed in the arbitrary location, should have taken care of modifying cap_frontier. That was my intuition, do you think I should change it?